sandbox/extract-dataset-metadata_emp.R

```diff
@@ -88,4 +89,324 @@ image_count = extract_image_count(exif)
 
 image_count
 
+#### file_size ####
+
+# Total file size of all images in set. Units: GB.
+
+# BEGIN FUNCTION CODE
+
+file_size = sum(exif$FileSize)
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_file_size = function (exif) {
+
+  file_size = sum(exif$FileSize) / 1000000000
+
+  return(file_size)
+
+}
+
+# test function on exif data
+
+file_size = extract_file_size(exif)
+
+file_size
+
+#### percent_images_rtk ####
+
+# Percent of images with an RTK fix. Use the EXIF RTKFlag. Will need to look up what the different codes mean.
+
+# BEGIN FUNCTION CODE
+
+exif$rtk_fix = extract_rtk_fix(exif)
+
+percent_images_rtk = (sum(exif$rtk_fix == TRUE) / nrow(exif)) * 100
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_percent_images_rtk = function (exif) {
+
+  exif$rtk_fix = extract_rtk_fix(exif)
+
+  percent_images_rtk = (sum(exif$rtk_fix == TRUE) / nrow(exif)) * 100
+
+  return(percent_images_rtk)
+
+}
+
+# test function on exif data
+
+percent_images_rtk = extract_percent_images_rtk(exif)
+
+percent_images_rtk
+
+
+#### white_balance_mode_derived ####
+
+# The most common white balance setting across all images in the project. Options: auto and manual
+
+# BEGIN FUNCTION CODE
+
+exif$white_balance = extract_white_balance(exif)
+
+Mode <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
+}
+
+white_balance_mode_derived = Mode(exif$white_balance)
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_white_balance_mode_derived = function (exif) {
+
+  exif$white_balance = extract_white_balance(exif)
+
+  Mode <- function(x) {
+    ux <- unique(x)
+    ux[which.max(tabulate(match(x, ux)))]
+  }
+
+  white_balance_mode_derived = Mode(exif$white_balance)
+
+  return(white_balance_mode_derived)
+
+}
+
+# test function on exif data
+
+white_balance_mode_derived = extract_white_balance_mode_derived(exif)
+
+white_balance_mode_derived
+
+#### white_balance_mode_prop_derived ####
+
+# The proportion of images with a white balance setting that matches the modal white balance setting in the dataset
+
+# BEGIN FUNCTION CODE
+
+exif$white_balance = extract_white_balance(exif)
+
+white_balance_mode_derived = extract_white_balance_mode_derived(exif)
+
+white_balance_mode_prop_derived = (sum(exif$white_balance == white_balance_mode_derived) / nrow(exif))
```

Comment:
I would merge the additional logic here into the function `extract_white_balance_mode_derived` to reduce redundancy, and have that function (which you should rename) return a list of two named values.